### PR TITLE
inspect: getmodulename takes a PathLike

### LINKED ISSUE #10328: inspect.getmodulename accepts pathlib.Path
Consider this simple snippet:

```python
# mod.py
import inspect
import pathlib

inspect.getmodulename(pathlib.PurePosixPath('foo.py'))
```

Run it with Python 3.8 or later and it runs fine (I don't have Python 3.7 due to ARM).

Check it with mypy and it fails:

```
 draft $ mypy mod.py
mod.py:4: error: Argument 1 to "getmodulename" has incompatible type "PurePosixPath"; expected "str"  [arg-type]
Found 1 error in 1 file (checked 1 source file)
```

Indeed the [definition](https://github.com/python/typeshed/blob/df795220a376ae9371afdc60dbf160392b93b6c2/stdlib/inspect.pyi#L180) indicates that only `str` is allowed for input.

The [implementation](https://github.com/python/cpython/blob/bc07c8f096791d678ca5c1e3486cb9648f7a027b/Lib/inspect.py#L936-L946) passes the input through to `os.path.basename`, which accepts `pathlib.Path`. Probably the typeshed should reflect this behavior, especially considering that the docs suggest that the input is a "path".

### PR DESCRIPTION
**Pull request by hauntsaninja**
Fixes #10328

### CODE REVIEW

File: stdlib/inspect.pyi [hunk 1]
```diff
@@ -177,7 +178,7 @@ if sys.version_info >= (3, 11):
     @overload
     def getmembers_static(object: object, predicate: _GetMembersPredicate | None = None) -> _GetMembersReturn: ...
 
-def getmodulename(path: str) -> str | None: ...
+def getmodulename(path: PathLike[str]) -> str | None: ...
```
Comment (AlexWaygood) [inline stdlib/inspect.pyi:181]: It also accepts `str`. We should use `_typeshed.StrPath`

```suggestion
def getmodulename(path: StrPath) -> str | None: ...
```

**Review by AlexWaygood — CHANGES_REQUESTED** [review]

File: stdlib/inspect.pyi (hunk 1 above)
Comment (hauntsaninja) [reply stdlib/inspect.pyi:181]: Oh oof, thanks for the review!

**Comment by jaraco** [comment]
> According to [mypy_primer](https://github.com/hauntsaninja/mypy_primer), this change has no effect on the checked open source code. 🤖🎉

There's code this will affect that's expected to land in `importlib_metadata` soon. Will that help?

**Review by AlexWaygood — APPROVED** [review]
LGTM

**Comment by AlexWaygood** [comment]
> > According to [mypy_primer](https://github.com/hauntsaninja/mypy_primer), this change has no effect on the checked open source code. 🤖🎉
> 
> There's code this will affect that's expected to land in `importlib_metadata` soon. Will that help?

It's okay, we agree that `inspect.getmodulename` takes path-like objects; I think we're good to go on this :-)

**Comment by hauntsaninja** [comment]
Thanks, I'll add importlib_metadata to the primer corpus! I was just a little sad my initial mistake wasn't flagged

**Comment by jaraco** [comment]
> Thanks, I'll add importlib_metadata to the primer corpus! I was just a little sad my initial mistake wasn't flagged

Good point. Does that mean that there are no uses in the primer of `getmodulename(str)`? That's a little surprising. The `importlib_metadata` changes won't help with that as they'll be passing a subclass of `pathlib.PurePosixPath`.

**Comment by jaraco** [comment]
Big thanks for the quick turnaround!

**Pull request merged**